rpc/src/event.rs

```diff
@@ -50,38 +32,21 @@ impl Event {
     }
 }
 
-impl<Ev> From<DialectEvent<Ev>> for Event
-where
-    Ev: Into<abci::Event>,
-{
-    fn from(msg: DialectEvent<Ev>) -> Self {
-        Event {
-            query: msg.query,
-            data: msg.data.into(),
-            events: msg.events,
-        }
-    }
-}
-
-impl<Ev> From<Event> for DialectEvent<Ev>
-where
-    abci::Event: Into<Ev>,
-{
-    fn from(msg: Event) -> Self {
-        DialectEvent {
-            query: msg.query,
-            data: msg.data.into(),
-            events: msg.events,
-        }
-    }
-}
-
 #[derive(Debug, Clone, PartialEq, Eq)]
-// To be fixed in 0.24
-#[allow(clippy::large_enum_variant)]
 pub enum EventData {
```

Comment:
I thought of making the `NewBlock` variant in this API type carry both 0.37 and 0.38 fields as optionals just like in `latest::DialectEventData`, and decoding to a different set of fields getting `Some` values for each respective version, but this feels like we'd be ramifying the enum matching that typically needs to be done by the consumer across multiple levels of the data structure.
So I decided a more clean split of the old and the new data is needed, also because we should eventually deprecate and remove `LegacyNewBlock` when everything up to 0.37 is EOLd.

Or do you have another idea about the API for this? Consider the dialect stuff in submodules an implementation detail for decoding JSON-RPC; I think the whole dialect business should be made private, the only reason I haven't made it private is the "integration" tests that decode fixtures, but those more properly belong in unit tests.

NB: Hermes currently does not care about the changed fields at all, it just needs `block` as can be seen in https://github.com/informalsystems/hermes/pull/3372.